util.js

```diff
@@ -33,7 +34,15 @@ exports.getAvaConfig = filepath => {
 
 	try {
 		const packageContent = JSON.parse(fs.readFileSync(filepath, 'utf8'));
-		return (packageContent && packageContent.ava) || defaultResult;
+
+		if (packageContent && packageContent.ava) {
+			return packageContent.ava;
+		}
+
+		const fileOptions = path.parse(filepath);
+		const avaConfig = esmRequire(path.join(fileOptions.dir, 'ava.config.js'));
+
+		return avaConfig.default;
```

Comment:
The config file may export a factory method that's invoked with an options argument. For here, that should be `{projectDir: fileOptions.dir}`.

We'll need to ignore `MODULE_NOT_FOUND` errors and return `defaultResult` instead.